src/View/Components/IncidentTimeline.php

```diff
@@ -49,15 +53,24 @@ private function incidents(Carbon $startDate, Carbon $endDate, bool $onlyDisrupt
                 'updates' => fn ($query) => $query->orderByDesc('created_at'),
             ])
             ->visible(auth()->check())
-            ->where(function (Builder $query) use ($endDate, $startDate) {
-                $query->whereBetween('occurred_at', [
-                    $endDate->startOfDay()->toDateTimeString(),
-                    $startDate->endofDay()->toDateTimeString(),
-                ])->orWhere(function (Builder $query) use ($endDate, $startDate) {
-                    $query->whereNull('occurred_at')->whereBetween('created_at', [
+            ->when($this->appSettings->recent_incidents_only, function ($query) {
+                $query->whereDate(
+                    'occurred_at',
+                    '>',
+                    Carbon::now()->subDays($this->appSettings->recent_incidents_days)->format('Y-m-d')
```

Comment:
I think this also needs to account for the `created_at` timestamp, when `occurred_at` is null.

Reply:
Ok, I will make necessary adjustments.